src/util/createInstantSearch.js

```diff
@@ -90,11 +90,12 @@ export const createInstantSearch = instantSearchOptions => {
   // we can then reuse that InstantSearch instance seamlessly from `ais-ssr`
   const rootMixin = {
     provide() {
+      const name = multiIndex ? `$_ais-${indexName}` : '$_ais';
```

Comment:
we could do this without the new option without breaking change

Reply:
I added this option because as the multi index need a custom ssr component, those who don't need it could keep using the original ssr component. If we remove this option, the original ssr component will stop working. Wouldn't that be a problem?

Reply:
Ah yes, I see the issue; the server side rendering component can no longer read from it, since it doesn't have access to the index name. This could be solved by making that a prop again, but as you pointed out, it would become a breaking change for users who use the current version

Reply:
Also we can't access props inside of inject. I tried.

Reply:
How are you configuring the component in your case then? You're overriding it in your custom ais-instant-search-ssr?

Reply:
Yes, as I exported the function **createInstantSearchComponent**, I'm able to create my own components and setting the injection name manually. It was the only way I found that works.

Reply:
Would be much better and easier with the prop, but unfortunately isn't possible.